src/containers/Transactions/useFetchHistory.ts

```diff
@@ -1,152 +1,83 @@
-    endDate: new Date('2023-09-20T00:00:00Z'),
-    cliffDate: new Date('2023-09-12T00:00:00Z'),
-    isCancellable: true,
-  },
-  {
-    model: 'exponential',
-    amountPerSecond: 12,
-    token: 'XRP',
-    from: '0x7890cdef1',
-    to: '0x234567890',
-    streamId: 4,
-    startDate: new Date('2023-09-15T00:00:00Z'), // ONGOING
-    endDate: new Date('2023-09-25T00:00:00Z'),
-    cliffDate: new Date('2023-09-18T00:00:00Z'),
-    isCancellable: false,
-  },
-  {
-    model: 'linear',
-    amountPerSecond: 15,
-    token: 'ADA',
-    from: '0x111111111',
-    to: '0x222222222',
-    streamId: 5,
-    startDate: new Date('2023-09-30T00:00:00Z'), // PENDING
-    endDate: new Date('2023-10-10T00:00:00Z'),
-    cliffDate: new Date('2023-10-03T00:00:00Z'),
-    isCancellable: true,
-  },
-];
-
-const fetchStreams = async (address: string): Promise<IStream[]> => {
-  return mockStreams;
+import BN from 'src/utils/BN';
+import fetch from 'src/utils/request';
+import { ExternalPages } from 'src/constants/externalPages';
+import { IResponseStreamsResult, IStreamHistory } from 'src/constants/types';
+
+const fetchStreams = async (address: string) => {
+  if (!address) {
+    return [];
+  }
+  const qsAddress = qs.stringify({ address });
+  try {
+    const streams = await fetch<IResponseStreamsResult>(
+      `${ExternalPages.FLUXITY_API}/testnet/stream?${qsAddress}`,
+      {
+        method: 'GET',
+        headers: { accept: 'application/json' },
+      },
+    );
+
+    return streams.data.result;
+  } catch (e) {}
+  return [];
 };
-const currentDate = new Date();
 
-function calculateCompletionPercentage(stream: IStream) {
-  const { startDate, endDate } = stream;
-  const currentDate = new Date().getTime();
+const calculateCompletionPercentage = (start_date: number, end_date: number) => {
+  const currentDate = Date.now();
+  const endDate = new Date(end_date).getTime() * 1000;
+  const startDate = new Date(start_date).getTime() * 1000;
 
-  if (currentDate < startDate.getTime()) {
+  if (currentDate < startDate) {
     return 0;
   }
 
-  if (currentDate >= endDate.getTime()) {
+  if (currentDate >= endDate) {
     return 100;
   }
 
-  const elapsedMilliseconds = currentDate - startDate.getTime();
-
-  const totalMilliseconds = endDate.getTime() - startDate.getTime();
+  const elapsedTime = new BN(currentDate).minus(startDate);
 
-  const completionPercentage = (elapsedMilliseconds / totalMilliseconds) * 100;
+  const totalTime = new BN(endDate).minus(startDate);
 
-  return completionPercentage.toFixed();
-}
+  const completionPercentage = new BN(elapsedTime).div(totalTime).times(100);
 
-function streamStatusHandler(stream: IStream) {
-  if (currentDate < stream.startDate) {
-    return StreamStatus.PENDING;
-  } else if (currentDate < stream.endDate) {
-    return StreamStatus.ONGOING;
-  }
-  return StreamStatus.EXPIRED;
-}
+  return completionPercentage.toFixed(0);
+};
 
+const calculateAmount = (amount: string) => {
```

Comment:
rename to `fromDecimals(amount, decimals = 7)` and move it to utils, next to `toDecimals`